src/entities/attendance-badge/api/mocks/modify-badge.ts

```diff
@@ -25,10 +25,34 @@ export const modifyBadgeMockHandler = createMockHandler({
     }
 
     ATTENDANCE_BADGE_MOCK_DATA[targetIndex] = {
-      ...ATTENDANCE_BADGE_MOCK_DATA[targetIndex],
-      ...data,
+      id: ATTENDANCE_BADGE_MOCK_DATA[targetIndex].id,
+      name: data.name,
+      icon: data.icon,
+      conditionGroups: data.conditionGroups.map((group) => ({
+        groupId:
+          Math.max(
+            0,
+            ...ATTENDANCE_BADGE_MOCK_DATA.flatMap(({ conditionGroups }) => {
+              return conditionGroups.map(({ groupId }) => groupId)
+            }),
+          ) + 1,
+        conditions: group.map((condition) => ({
+          id:
+            Math.max(
+              0,
+              ...ATTENDANCE_BADGE_MOCK_DATA.flatMap((badge) => {
+                return badge.conditionGroups.flatMap(({ conditions }) => {
+                  return conditions.map(({ id }) => id)
+                })
+              }),
+            ) + 1,
+          ...condition,
+        })),
+      })),
```

Comment:
[제안]
이 부분에서 최상위 map 함수의 형태와 같이 모두 return을 제거하는 것은 어떠신가요? 일관성과 가독성을 모두 챙겨갈 수 있을 것 같아요.

```suggestion
      conditionGroups: data.conditionGroups.map((group) => ({
        groupId:
          Math.max(
            0,
            ...ATTENDANCE_BADGE_MOCK_DATA.flatMap(({ conditionGroups }) => 
              conditionGroups.map(({ groupId }) => groupId)
            ),
          ) + 1,
        conditions: group.map((condition) => ({
          id:
            Math.max(
              0,
              ...ATTENDANCE_BADGE_MOCK_DATA.flatMap((badge) => 
                badge.conditionGroups.flatMap(({ conditions }) => conditions.map(({ id }) => id)),
              ),
            ) + 1,
          ...condition,
        })),
      })),
```

Reply:
현재 리스트 프로세싱 메서드 내 콜백 함수에 대한 컨벤션이 없어 문제가 되는 것 같습니다. 추후 리팩터링 스프린트 때 해당 부분 반영해보겠습니다.